ngraph/core/include/openvino/core/tensor.hpp

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief This is a header file for the OpenVINO Runtime tensor API
+ *
+ * @file openvino/runtime/tensor.hpp
+ */
+#pragma once
+
+#include "openvino/runtime/common.hpp"
+#include "openvino/core/core_visibility.hpp"
+#include "openvino/core/shape.hpp"
+#include "openvino/core/type/element_type.hpp"
+#include "openvino/core/coordinate.hpp"
+
+namespace InferenceEngine {
+class Blob;
+}  // namespace InferenceEngine
+
+namespace ov {
+namespace runtime {
+class InferRequest;
+class RemoteTensor;
+class RemoteContext;
+class VariableState;
+}  // namespace runtime
+
+/**
+ * @brief Memory access and interpretation API
+ *
+ * It can throw exceptions safely for the application, where it is properly handled.
+ */
+class OPENVINO_API Tensor {
+    std::shared_ptr<void>       _so;
+    std::shared_ptr<ie::Blob>   _impl;
+
+    /**
+     * @brief Constructs Tensor from the initialized std::shared_ptr
+     * @param so Plugin to use. This is required to ensure that Tensor can work properly even if plugin object is
+     * destroyed.
+     * @param impl Initialized shared pointer
+     */
+    Tensor(const std::shared_ptr<void>& so,
+           const std::shared_ptr<ie::Blob>& impl);
+
+    friend class ov::runtime::InferRequest;
+    friend class ov::runtime::RemoteTensor;
+    friend class ov::runtime::RemoteContext;
+    friend class ov::runtime::VariableState;
+
+public:
+    /**
+     * @brief Default constructor
+     */
+    Tensor() = default;
+
+    /**
+     * @brief Constructs Tensor using element type and shape. Allocate internal host storage using default allocator
+     * @param type Tensor element type
+     * @param shape Tensor shape
+     */
+    Tensor(const element::Type type, const Shape& shape);
+
+    /**
+     * @brief Constructs Tensor using element type and shape. Wraps allocated host memory.
+     * @note Dose not own the memory
+     * @param type Tensor element type
+     * @param shape Tensor shape
+     * @param ptr Pointer to allocate memory
+     */
+    Tensor(const element::Type type, const Shape& shape, void* ptr);
+
+    Tensor(const Tensor& other,
+           const Coordinate& begin,
+           const Coordinate& end,
+           const Strides& strides = {},
+           const element::Type type = {});
```

Comment:
1. remove strides and type here
2. add strides to the previous constructor
3. I don't see tests for these constructors

@ilyachur how do you think?

Reply:
done